### PR TITLE
add 'env-var' meta

### PR DESCRIPTION
**Pull request by DrewDennison**

### CODE REVIEW

**Review by underyx — APPROVED** [review]

File: src/semgrep_agent/meta.py [hunk 1]
```diff
@@ -113,6 +113,100 @@ def to_dict(self) -> Dict[str, Any]:
...
+
+    @cachedproperty
+    def commit_title(self) -> Optional[str]:
+        return os.getenv("SEMGREP_COMMIT_TITLE")
+
+    @cachedproperty
+    def pr_title(self) -> Optional[str]:
+        return os.getenv("SEMGREP_PR_TITLE")
+
+    def to_dict(self) -> Dict[str, Any]:
```
Comment (chmccreery) [inline src/semgrep_agent/meta.py:187]: Is there a way to get around this being a duplicate of the GitMeta to_dict? I could see us updating one and forgetting to update the other/ seems like a good place for code deduplication

**Review by chmccreery — APPROVED** [review]

File: src/semgrep_agent/meta.py (hunk 1 above)
Comment (DrewDennison) [reply src/semgrep_agent/meta.py:187]: Yeah I should fold these into the GitMeta class on reflection. Thanks for the feedback

**Comment by chmccreery** [comment]
@DrewDennison do you want to close this or fold the changes into GitMeta?

**Comment by DrewDennison** [comment]
closing for now and I'll add into GitMeta as tech debt

**Pull request closed without merging**